src/main/scala/it/pagopa/interop/agreementprocess/service/AgreementStateByAttributesFSM.scala

```diff
@@ -0,0 +1,62 @@
+package it.pagopa.interop.agreementprocess.service
+
+import it.pagopa.interop.agreementmanagement.client.model.AgreementState
+import it.pagopa.interop.agreementmanagement.client.model.AgreementState._
+import it.pagopa.interop.catalogmanagement.client.model.{Attribute, EService}
+import it.pagopa.interop.tenantmanagement.client.model.Tenant
+
+import java.util.UUID
+
+object AgreementStateByAttributesFSM {
+
+  def nextState(currentState: AgreementState, eService: EService, consumer: Tenant): AgreementState =
+    currentState match {
+      case DRAFT                        =>
+        if (!certifiedAttributesSatisfied(eService, consumer)) MISSING_CERTIFIED_ATTRIBUTES
+        else if (declaredAttributesSatisfied(eService, consumer)) PENDING
+        else DRAFT
+      case PENDING                      =>
+        if (!certifiedAttributesSatisfied(eService, consumer)) MISSING_CERTIFIED_ATTRIBUTES
+        else if (!declaredAttributesSatisfied(eService, consumer)) DRAFT
+        else if (!verifiedAttributesSatisfied(eService, consumer)) PENDING
+        else ACTIVE
+      case ACTIVE                       =>
+        if (
+          certifiedAttributesSatisfied(eService, consumer) &&
+          declaredAttributesSatisfied(eService, consumer) &&
+          verifiedAttributesSatisfied(eService, consumer)
+        )
+          ACTIVE
+        else
+          SUSPENDED
+      case SUSPENDED                    =>
+        if (
+          certifiedAttributesSatisfied(eService, consumer) &&
+          declaredAttributesSatisfied(eService, consumer) &&
+          verifiedAttributesSatisfied(eService, consumer)
+        )
+          ACTIVE
+        else
+          SUSPENDED
+      case ARCHIVED                     => ARCHIVED
+      case MISSING_CERTIFIED_ATTRIBUTES =>
+        if (certifiedAttributesSatisfied(eService, consumer)) DRAFT
+        else MISSING_CERTIFIED_ATTRIBUTES
+    }
+
+  def certifiedAttributesSatisfied(eService: EService, consumer: Tenant): Boolean =
+    attributesSatisfied(eService.attributes.certified, consumer.attributes.flatMap(_.certified).map(_.id))
+
+  def declaredAttributesSatisfied(eService: EService, consumer: Tenant): Boolean =
+    attributesSatisfied(eService.attributes.declared, consumer.attributes.flatMap(_.declared).map(_.id))
+
+  def verifiedAttributesSatisfied(eService: EService, consumer: Tenant): Boolean =
+    attributesSatisfied(eService.attributes.verified, consumer.attributes.flatMap(_.verified).map(_.id))
+
+  private def attributesSatisfied(requested: Seq[Attribute], assigned: Seq[UUID]): Boolean =
+    requested.forall {
+      case Attribute(Some(single), _) => assigned.contains(single.id)
+      case Attribute(_, Some(group))  => group.map(_.id).intersect(assigned).nonEmpty
+      case _                          => true
+    }
```

Comment:
Which case is described here? 
An attribute could be either single or group not both at the same time.
What do you think about the solution below? (Maybe I'm missing some other details).
```
    requested.forall {
      case Attribute(Some(single), None) => assigned.contains(single.id)
      case Attribute(None, Some(group))  => group.map(_.id).intersect(assigned).nonEmpty
      case _                          => false // better an error
    }
```

Reply:
I was going for the optimistic approach, but you're right. What about this one?
```suggestion
    requested.forall {
      case Attribute(Some(single), _)  => assigned.contains(single.id)
      case Attribute(_, Some(group))   => group.map(_.id).intersect(assigned).nonEmpty
      case Attribute(Some(_), Some(_)) => false
      case Attribute(None, None)       => true
    }
```
this way we can ignore wrong but "empty" data, but block potential dangerous configurations

Reply:
@beetlecrunch or do you prefer to always fail on an unexpected combination?

Reply:
I think failure is the best choice, although it is almost impossible. 
I open a task for this improvement, for now, your suggestion is acceptable.

Reply:
I can keep the failure, no problem.
BTW Ruggero suggested IMO a very good refactor for this model that would simplify a lot of logics.
But let's discuss it another time

Reply:
rollback: if we set default to `false` it means that in case of internal errors (i.e. both `single` and `group` have values) the platform would suspend/block an agreement.
So we agreed to keep the default to `true`